lib/stimulus_reflex/element.rb

```diff
@@ -3,11 +3,12 @@
 class StimulusReflex::Element < OpenStruct
   attr_reader :attributes, :data_attributes
 
-  def initialize(attrs = {})
-    @attributes = HashWithIndifferentAccess.new(attrs || {})
-    @data_attributes = attributes.each_with_object(HashWithIndifferentAccess.new) { |(key, value), memo|
-      memo[key.delete_prefix("data-")] = value if key.start_with?("data-")
-    }
+  def initialize(data = {})
+    @attributes = HashWithIndifferentAccess.new(data["attrs"] || {})
+    @data_attributes = (data["dataset"] || {}).each_with_object(HashWithIndifferentAccess.new) { |(key, value), memo|
+      memo[key.delete_prefix("data-")] = value
+    }.freeze
+
     super attributes.merge(attributes.transform_keys(&:underscore))
```

Comment:
I'm thinking that I'd prefer to include the combined dataset in the dot notation API available on the element. We'd need to change the call to `super`:

```ruby
super attributes.merge(attributes.merge(data_attributes).transform_keys(&:underscore))
```

Reply:
Isn't this already happening on line 16?

Maybe we could move that inside to constructor and get rid of the `dataset` accessor method

Reply:
No this is for constructing the `StimulusReflex::Element` as an `OpenStruct`. Note the call to `super`

Also, I prefer dataset get lazily initialized to help with performance if/when it isn't used.